src/NuGet.Core/NuGet.PackageManagement/BuildIntegratedPackageReference.cs

```diff
@@ -24,8 +25,8 @@ public class BuildIntegratedPackageReference : PackageReference
         /// Create a PackageReference based on a LibraryDependency.
         /// </summary>
         /// <param name="dependency">Full PackageReference metadata.</param>
-        public BuildIntegratedPackageReference(LibraryDependency dependency, NuGetFramework projectFramework)
-            : base(GetIdentity(dependency),
+        public BuildIntegratedPackageReference(LibraryDependency dependency, NuGetFramework projectFramework, PackageIdentity installedVersion)
```

Comment:
If you rebase, this is likely to complain about a breaking public API. 

I think you'll want to add an overload here.